src/rcheevos/operand.c

```diff
@@ -368,113 +374,15 @@ unsigned rc_evaluate_operand(rc_operand_t* self, rc_peek_t peek, void* ud, lua_S
       break;
 
     case RC_OPERAND_ADDRESS:
-    case RC_OPERAND_DELTA:
-    case RC_OPERAND_PRIOR:
-      switch (self->size) {
-        case RC_OPERAND_BIT_0:
-          value = (peek(self->value, 1, ud) >> 0) & 1;
-          break;
-
-        case RC_OPERAND_BIT_1:
-          value = (peek(self->value, 1, ud) >> 1) & 1;
-          break;
-        
-        case RC_OPERAND_BIT_2:
-          value = (peek(self->value, 1, ud) >> 2) & 1;
-          break;
-        
-        case RC_OPERAND_BIT_3:
-          value = (peek(self->value, 1, ud) >> 3) & 1;
-          break;
-        
-        case RC_OPERAND_BIT_4:
-          value = (peek(self->value, 1, ud) >> 4) & 1;
-          break;
-        
-        case RC_OPERAND_BIT_5:
-          value = (peek(self->value, 1, ud) >> 5 ) & 1;
-          break;
-        
-        case RC_OPERAND_BIT_6:
-          value = (peek(self->value, 1, ud) >> 6) & 1;
-          break;
-        
-        case RC_OPERAND_BIT_7:
-          value = (peek(self->value, 1, ud) >> 7) & 1;
-          break;
-
-        case RC_OPERAND_LOW:
-          value = peek(self->value, 1, ud) & 0x0f;
-          break;
-                
-        case RC_OPERAND_HIGH:
-          value = (peek(self->value, 1, ud) >> 4) & 0x0f;
-          break;
-        
-        case RC_OPERAND_8_BITS:
-          value = peek(self->value, 1, ud);
-
-          if (self->is_bcd) {
-            value = ((value >> 4) & 0x0f) * 10 + (value & 0x0f);
-          }
-
-          break;
-
-        case RC_OPERAND_16_BITS:
-          value = peek(self->value, 2, ud);
-
-          if (self->is_bcd) {
-            value = ((value >> 12) & 0x0f) * 1000
-                  + ((value >>  8) & 0x0f) *  100
-                  + ((value >>  4) & 0x0f) *   10
-                  + ((value >>  0) & 0x0f) *    1;
-          }
-
-          break;
-
-        case RC_OPERAND_24_BITS:
-          value = peek(self->value, 4, ud);
-
-          if (self->is_bcd) {
-            value = ((value >> 20) & 0x0f) * 100000
-                  + ((value >> 16) & 0x0f) *  10000
-                  + ((value >> 12) & 0x0f) *   1000
-                  + ((value >>  8) & 0x0f) *    100
-                  + ((value >>  4) & 0x0f) *     10
-                  + ((value >>  0) & 0x0f) *      1;
-          }
-
-          break;
-
-        case RC_OPERAND_32_BITS:
-          value = peek(self->value, 4, ud);
-
-          if (self->is_bcd) {
-            value = ((value >> 28) & 0x0f) * 10000000
-                  + ((value >> 24) & 0x0f) *  1000000
-                  + ((value >> 20) & 0x0f) *   100000
-                  + ((value >> 16) & 0x0f) *    10000
-                  + ((value >> 12) & 0x0f) *     1000
-                  + ((value >>  8) & 0x0f) *      100
-                  + ((value >>  4) & 0x0f) *       10
-                  + ((value >>  0) & 0x0f) *        1;
-          }
-
-          break;
-      }
+      value = self->memref->value;
+      break;
```

Comment:
When processing the conditions, we just have to look at the already updated memory references.